packages/build/src/plugins_core/pre_dev_cleanup/index.ts

```diff
@@ -18,18 +19,15 @@ const dirExists = async (path: string): Promise<boolean> => {
 const getDirtyDirs = async function ({
   buildDir,
   constants: { INTERNAL_EDGE_FUNCTIONS_SRC, INTERNAL_FUNCTIONS_SRC },
+  packagePath,
 }: CoreStepFunctionArgs): Promise<string[]> {
-  const dirs: string[] = []
+  const directories: string[] = [
+    resolve(buildDir, INTERNAL_FUNCTIONS_SRC),
+    resolve(buildDir, INTERNAL_EDGE_FUNCTIONS_SRC),
+    ...getBlobsDirs(buildDir, packagePath),
+  ]
 
-  if (INTERNAL_FUNCTIONS_SRC && (await dirExists(resolve(buildDir, INTERNAL_FUNCTIONS_SRC)))) {
-    dirs.push(INTERNAL_FUNCTIONS_SRC)
-  }
-
-  if (INTERNAL_EDGE_FUNCTIONS_SRC && (await dirExists(resolve(buildDir, INTERNAL_EDGE_FUNCTIONS_SRC)))) {
-    dirs.push(INTERNAL_EDGE_FUNCTIONS_SRC)
-  }
-
-  return dirs
+  return directories.filter(async (directory) => directory && (await dirExists(directory)))
```

Comment:
I believe this will be an `Promise<void>[]`, instead of an `Promise<string[]>`. Don't think an async function in `.filter` works. Maybe you could get an array of booleans using `.map` and `Promise.all` first, and filter in a second step?